### DISCUSSION TITLE
Partial rebuilds

### DISCUSSION THREAD

**Discussion by anuejn** [apertus-open-source-cinema/nmigen-gateware#14] [Design]
It would be nice, if the gateware wouldnt be rebuild if only pydriver changed.
It is currently unclear to me how to do that in a clean way

**Comment by rroohhh** [comment]
I think this could potentially be solved by by using `amigen`.

`amigen` more cleanly seperates extra information from the actual gateware. 

What `amigen` does not currently really have is a structured way to generate extra parts belonging to the gateware, but not part of the actual gateware (like the pydriver files). I think we should think of how we want to API there to look like. Currently `amigen` is used by simply converting everything into a `nmigen` `Module` in the end. Maybe we want to additionally output some custom information generated by the various `amigen` `Element`s? Or maybe we want to add our own custom build plan system?
